spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,359 @@
+    end
+
+    it "should raise if not a null value" do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "hello"
+      end
+    end
+  end
+
+  describe "Bool.from_yaml" do
+    describe "true values" do
+      it "should serialize and deserialize" do
+        Bool.from_yaml(true.to_yaml).should eq true
+      end
+
+      values = {"true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"}
+
+      values.each do |value|
+        it %(should return true if "#{value}") do
+          Bool.from_yaml(value).should eq true
+        end
+
+        it %(should raise if "#{value}" is quoted with double quotes) do
+          expect_raises(YAML::ParseException) do
+            Bool.from_yaml %("#{value}")
+          end
+        end
+
+        it %(should raise if "#{value}" is quoted with single quotes) do
+          expect_raises(YAML::ParseException) do
+            Bool.from_yaml "'#{value}'"
+          end
+        end
+      end
+    end
+  end
+
+  describe "false values" do
+    it "should serialize and deserialize" do
+      Bool.from_yaml(false.to_yaml).should eq false
+    end
+
+    values = {"false", "False", "FALSE", "off", "Off", "OFF", "n", "N", "no", "No", "NO"}
+
+    values.each do |value|
+      it %(should return false if "#{value}") do
+        Bool.from_yaml(value).should eq false
+      end
+
+      it %(should raise if "#{value}" is quoted with double quotes) do
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml %("#{value}")
+        end
+      end
+
+      it %(should raise if "#{value}" is quoted with single quotes) do
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml "'#{value}'"
+        end
+      end
+    end
+  end
+
+  it "should raise if not a bool value" do
+    expect_raises(YAML::ParseException) do
+      Bool.from_yaml "hello"
+    end
+  end
+end
+
+[Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64].each do |type|
+  describe "#{type.name}.from_yaml" do
+    it "should serialize and deserialize" do
+      type.from_yaml(type.new("1").to_yaml).should eq type.new("1")
+    end
+
+    it "should parse a number into an #{type.name}" do
+      type.from_yaml(type.new("1").to_yaml).should eq type.new("1")
+    end
+
+    it "should raise if quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml %("1")
+      end
+    end
+
+    it "should raise if quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml "'1'"
+      end
+    end
+
+    it "should raise when not a number" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml("hi")
+      end
+    end
+  end
+end
+
+[Float32, Float64].each do |type|
+  describe "#{type.name}.from_yaml" do
+    it "should serialize and deserialize" do
+      type.from_yaml(type.new("1.0").to_yaml).should eq type.new("1.0")
+    end
+
+    it "should parse a number into an {{type.id}}" do
```

Comment:
Should this be `it "should parse a number into an #{type.name}" do` , like on line 123?